columnflow/tasks/framework/base.py

```diff
@@ -447,12 +448,10 @@ def has_obj(name):
                 lookup.extend(list(object_groups[name]))
             elif accept_patterns:
                 # must eventually be a pattern, perform an object traversal
-                # make this also viable for multi-dim variables for plotting/histograms
-                name_parts = name.split("-")
-                all_objects = get_all_object_names()
-                # match all name parts
-                if all([law.util.multi_match(n, all_objects) for n in name_parts]):
-                    object_names.append(name)
+                for _name in sorted(get_all_object_names()):
```

Comment:
the problem is that this will not work anymore for multi dimensional variables

Reply:
I have changed it myself like this at some point before
```suggestion
                name_parts = name.split("-")
                # match all name parts
                matches = [[] for _ in name_parts]
                for i, n in enumerate(name_parts):
                    for _name in sorted(get_all_object_names()):
                        if law.util.multi_match(_name, n):
                            matches[i].append(_name)
                for matches_pair in itertools.product(*matches):
                    object_names.append("-".join(matches_pair))
```